docs/source/debugging.rst

```diff
@@ -0,0 +1,73 @@
+Debugging Models
+****************
+
+A number of errors and warnings may be raised when attempting to solve a model. Because GPkit doesn't let you build models that aren't geometric programs or sequential geometric programs, and checks the solver-returned result to ensure that the problem has actually been solved, these errors can generally be classed into two types: models where there is no possible solution (primal infeasible models) and those where nothing stops the objective (or variables in the objective) from going to 0 or infinity (dual infeasible models).
+
+..  Add section like the below once that code has made it to master:
+    GPkit contains several tools for diagnosing which constraints and variables might be causing this infeasibility. The first thing to try with a GP that doesn't solve is running `m.debug()`,
+    which will say what combinations of changes would make the model feasible. Of course, certain modeling errors (such as forgetting a constraint) will be harder to determine from this output than others. (NOTE: that speaks to a whole other modeling-process section that could be written...) Because SPs and sequential geometric programs do not have the same guarantees as geometric programs, the construction of feasibility models for them should be done by hand, as discussed below. (NOTE: add Relax to the discussion below as well)
+
+
+Potential errors and warnings
+=============================
+
+- ``RuntimeWarning: Primal solution violates constraint: 1.0000149786 is greater than 1``
+    - this warning may be seen in dual infeasible models, see *Dual Infeasibility* below.
+
+- ``RuntimeWarning: Dual cost nan does not match primal cost 1.00122315152``
+    - this warning may be seen in dual infeasible models, see *Dual Infeasibility* below.
+
+- ``RuntimeWarning: final status of solver 'cvxopt' was 'unknown', not 'optimal’`` or ``RuntimeWarning: final status of solver 'mosek' was ‘UNKNOWN’, not 'optimal’.``
+    - this is the most difficult warning to debug. It can be thrown when attempting to solve a dual infeasible model or a primal infeasible model. See *Dual* and *Primal Infeasibility* below.
+
+- ``RuntimeWarning: final status of solver 'mosek' was 'DUAL_INFEAS_CER', not 'optimal’``
+    - this error is thrown when attempting to solve a dual infeasible model with MOSEK,  see *Dual Infeasibility* below.
+
+- ``RuntimeWarning: final status of solver 'mosek' was 'PRIM_INFEAS_CER', not 'optimal’``
+    - this error is thrown when attempting to solve a primal infeasible model with MOSEK,  see *Primal Infeasibility* below.
+
+
+Dual Infeasibility
+==================
+
+In some cases a model will not solve because variables are being pushed by the objective to 0 or infinity. If the solver catches such behaviour it will return ``dual infeasible`` (or equivalent), but sometimes solvers do not catch it and return ``unknown``.
```

Comment:
This could be made more precise. Proposal below.

```
In some cases a model will not solve because the optimal value of one or more variables is 0 or infinity (plus or minus infinity in logspace). Such a problem is known as ``dual infeasible`` because the GP's dual problem, which determines the optimal values of the sensitivites, does not have any feasible solution. If the solver can prove that the dual is infeasible, it will return a dual infeasibility certificate. Otherwise, it may finish with a solution status of ``unknown``.